metals/src/main/scala/scala/meta/internal/metals/BSPErrorHandler.scala

```diff
@@ -0,0 +1,90 @@
+package scala.meta.internal.metals
+
+import java.nio.file.Files
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import scala.util.control.NonFatal
+
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.metals.clients.language.MetalsLanguageClient
+import scala.meta.io.AbsolutePath
+
+import org.eclipse.{lsp4j => l}
+
+class BSPErrorHandler(
+    languageClient: MetalsLanguageClient,
+    workspaceFolder: AbsolutePath,
+)(implicit context: ExecutionContext) {
+
+  private lazy val logsPath = workspaceFolder.resolve(Directories.log)
+
+  def showError(message: String): Future[Unit] = Future {
+    BSPErrorHandler.makeShowMessage(message) match {
+      case Right(params) =>
+        scribe.error(message)
+        languageClient.showMessage(params)
+      case Left(params) =>
+        val lastLine = getLastLogsLine().map(_ + 1).getOrElse(0)
+        scribe.error(message)
+        languageClient.showMessageRequest(params).asScala.flatMap { item =>
+          Option(item).map(_.getTitle()) match {
+            case Some(BSPErrorHandler.goToLogs) =>
+              Future(gotoLogs(lastLine))
```

Comment:
We should probably look for the last line with the error, since there might have been more messages since then.